### ISSUE TITLE
Oracle call uses To_Timestamp not To_Date

### ISSUE THREAD

**Issue by sarabu07** [linq2db/linq2db#431]
```cs
var pollTime = DateTime.Now.AddMinutes(-15);  

var contracts = (from c in dbConnection.GetTable<Contract>()
                                     where c.Last_Modified > pollTime
                                     select c.Contract_ID);
```
The above linq creates following sql

```sql
--  Oracle.Managed Oracle
DECLARE @pollTime TimeStamp -- DateTime
SET     @pollTime = TO_TIMESTAMP('2016-09-08 10:42:31.2056108', 'YYYY-MM-DD HH24:MI:SS.FF7')

SELECT
               t1.Contract_ID
FROM
               Contract t1
WHERE
               t1.Last_Modified > :pollTime
```

**Comment by sarabu07** [comment]
is there a way we can make the query to use To_Date function in oracle from linq2db?

**Comment by ili** [comment]
You can try this:
```cs
public class Contract
{
    [Column(DataType = DataType.Date]
    pablic DateTime Last_Modified ;
}
```

But in general parameters are only printed for informational purpose. There are no this text in real query send to servet, you can check `DataConnection.LastQuery` for example. Parameters are formed and filled by data clients.

**Comment by ili** [comment]
Closing, will reopen on additional info

In general declaring & assigning parameters are shown only in debug for informational purpose.

Only select closure is sent to the server. So assigning and sending parameters is not under linq2db, Oracle provider is responsible for it.

**Comment by filippobottega** [comment]
I have the same problem: I'm not able to use dates in queries because Oracle provider uses To_Timestamp instead of To_Date.  Using `[Column(DataType = DataType.Date]` doesn't solve the issue.
Which type of additional info do you need? do you need a sample code to replicate the behavior?

**Comment by filippobottega** [comment]
As a workaround I have create a Sql Expression to cast TIMESTAMP to DATE:

``` C#
   public static  class SqlExpressions
    {
        [Sql.Expression("CAST({0} AS DATE)", ServerSideOnly = true)]
        public static DateTime CastAsDate(this DateTime x)
        {
            throw new InvalidOperationException();
        }
    }
```

You can rewrite the query as:

``` C#
var pollTime = DateTime.Now.AddMinutes(-15);  

var contracts = (from c in dbConnection.GetTable<Contract>()
                                     where c.Last_Modified > pollTime.CastAsDate()
                                     select c.Contract_ID);
```

**Comment by MaceWindu** [comment]
Closed by #2058
